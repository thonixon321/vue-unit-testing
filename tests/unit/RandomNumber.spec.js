import RandomNumber from '@/components/RandomNumber'
import { mount } from "@vue/test-utils";
//TESTING PROPS
describe('RandomNumber', () => {
  test('By default, randomNumber data value should be 0', () => {
    const wrapper = mount(RandomNumber)
    //the span contains the randomNumber data value - should be 0 on load
    expect(wrapper.html()).toContain('<span>0</span>')
  })

  test('If button is clicked, randomNumber should be between 1 and 10', async () => {
    const wrapper = mount(RandomNumber)
    wrapper.find('button').trigger('click')
    //have to wait for dom to finish rendering so do the await $nextTick
    await wrapper.vm.$nextTick()
    //after button is clicked, check the text in span
    const randomNumber = parseInt(wrapper.find('span').element.textContent)
    expect(randomNumber).toBeGreaterThanOrEqual(1)
    expect(randomNumber).toBeLessThanOrEqual(10)
  })

  test('If button is clicked, randomNumber should be between 200 and 300', async () => {
    //this is how you update the props in component (200 for min, 300 for max)
    const wrapper = mount(RandomNumber, {
      propsData: {
        min: 200,
        max: 300,
      }
    })
    wrapper.find('button').trigger('click')
    //have to wait for dom to finish rendering so do the await $nextTick
    await wrapper.vm.$nextTick()
    //after button is clicked, check the text in span
    const randomNumber = parseInt(wrapper.find('span').element.textContent)
    expect(randomNumber).toBeGreaterThanOrEqual(200)
    expect(randomNumber).toBeLessThanOrEqual(300)
  })
})