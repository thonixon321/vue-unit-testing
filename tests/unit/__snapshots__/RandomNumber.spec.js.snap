// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomNumber If button is clicked, randomNumber should be between 200 and 300 1`] = `<div><span>273</span> <button>Generate Random Number</button></div>`;
